src/Maestro/tests/Maestro.ScenarioTests/MaestroScenarioTestBase.cs

```diff
@@ -323,5 +449,87 @@ public async Task<string> GetRepositoryPolicies(string repoUri, string branchNam
+                bool checkDir = Directory.Exists(path);
+                Assert.IsTrue(Directory.Exists(path), $"Expected cloned repo '{name}' but not found at '{path}'");
+
+                string versionDetailsPath = Path.Combine(path, "eng", "Version.Details.xml");
+                if (File.Exists(versionDetailsPath))
+                {
+                    using (FileStream stream = File.OpenRead(versionDetailsPath))
+                    {
+                        SHA256 Sha256 = SHA256.Create();
+                        byte[] hash = Sha256.ComputeHash(stream);
+                        string hashString = BitConverter.ToString(hash);
+                        Assert.AreEqual(hashString, repos[name], $"Expected {versionDetailsPath} to have hash '{repos[name]}', actual hash '{hashString}'");
+                    }
+                }
+                else
+                {
+                    Assert.AreEqual(repos[name], "", $"Expected a '{versionDetailsPath}' but it is missing");
+                }
+            }
+
+            foreach (string repo in masterRepos)
+            {
+                string repoPath = Path.Combine(reposFolder, repo);
+                Assert.IsTrue(Directory.Exists(repoPath), $"Expected cloned master repo '{repo}' but not found at '{repoPath}'");
+
+                string gitRedirectPath = Path.Combine(repoPath, ".git");
+                string expectedGitDir = Path.Combine(gitDirFolder, repoPath);
+                string expectedRedirect = $"gitdir: {expectedGitDir}.git";
+                string actualRedirect = Directory.GetFiles(gitRedirectPath).First();
+                StringAssert.AreEqualIgnoringCase(expectedRedirect, actualRedirect, $"Expected '{repoPath}' to have a .gitdir redirect of '{expectedRedirect}', actual '{actualRedirect}'");
+            }
+
+            string[] actualRepos = Directory.GetDirectories(TestContext.CurrentContext.TestDirectory, ".");
+            string[] actualMasterRepos = Directory.GetDirectories(TestContext.CurrentContext.TestDirectory).Where(dir => !dir.Contains(".")).ToArray();
+
+            foreach (string repo in actualRepos)
+            {
+                Assert.Contains(repo, repos.Keys, $"Found unexpected repo folder '{repo}'");
+            }
+
+            foreach (string repo in actualMasterRepos)
+            {
+                Assert.Contains(repo, masterRepos, $"Found unexpected master repo folder '{repo}'");
+            }
+
+            foreach (string gitDir in gitDirs)
+            {
+                string gdPath = Path.Combine(gitDirFolder, gitDir);
+                Assert.IsTrue(Directory.Exists(gdPath), $"Expected a .gitdir for '{gitDir}' but not found at '{gdPath}'");
+            }
+
+            string[] actualGitDirs = Directory.GetDirectories(gitDirFolder);
+
+            foreach (string gitDir in actualGitDirs)
+            {
+                Assert.Contains(gitDir, actualGitDirs, $"Found unexpected .gitdir '{gitDir}'");
```

Comment:
... This is asserting that the thing we are iterating over contains the things inside it...